mne/preprocessing/maxwell.py

```diff
@@ -41,6 +41,136 @@
+    %(verbose)s
+
+    Returns
+    -------
+    raw_er_prepared : instance of Raw
+        A copy of the passed empty-room recording, ready for Maxwell filtering.
+
+    Notes
+    -----
+    This function will:
+
+    * Compile the list of bad channels according to the ``bads`` parameter.
+    * Inject the device-to-head transformation matrix from the experimental
+      recording into the empty-room recording.
+    * Set the following properties of the empty-room recording to match the
+      experimental recording:
+
+      * Montage
+      * ``raw.first_time`` and ``raw.first_samp``
+
+    * Adjust annotations according to the ``annotations`` parameter.
+    * Adjust the measurement date according to the ``meas_date`` parameter.
+
+    .. versionadded:: 1.1
+    """  # noqa: E501
+    _validate_type(item=raw_er, types=BaseRaw, item_name='raw_er')
+    _validate_type(item=raw, types=BaseRaw, item_name='raw')
+    _validate_type(item=bads, types=(list, str), item_name='bads')
+    if isinstance(bads, str):
+        _check_option(
+            parameter='bads', value=bads,
+            allowed_values=['from_raw', 'union', 'keep']
+        )
+    _validate_type(item=annotations, types=str, item_name='annotations')
+    _check_option(
+        parameter='annotations', value=annotations,
+        allowed_values=['from_raw', 'keep']
+    )
+    _validate_type(item=meas_date, types=str, item_name='meas_date')
+    _check_option(
+        parameter='meas_date', value=annotations,
+        allowed_values=['from_raw', 'keep']
+    )
+
+    raw_er_prepared = raw_er.copy()
+    del raw_er  # just to be sure
+
+    # handle bads; only keep MEG channels
+
+    if bads == 'from_raw':
+        bads = raw.info['bads']
+    elif bads == 'union':
+        bads = sorted(
+            set(raw.info['bads'] + raw_er_prepared.info['bads'])
+        )
+    elif bads == 'keep':
+        bads = raw_er_prepared.info['bads']
+
+    bads = [ch_name for ch_name in bads
+            if ch_name.startswith('MEG')]
+    raw_er_prepared.info['bads'] = bads
+
+    # handle dev_head_t
+    raw_er_prepared.info['dev_head_t'] = raw.info['dev_head_t']
+
+    # handle montage
+    montage = raw.get_montage()
+    raw_er_prepared.set_montage(montage)
+
+    # handle first_samp
+    raw_er_prepared._cropped_samp = raw._cropped_samp
+
+    # handle annotations & measurement date
+    if annotations == 'from_raw':
+        if meas_date == 'keep':
+            annotations = raw.annotations.copy()
+            annotations._orig_time = raw_er_prepared.info['meas_date']
+            raw_er_prepared.set_annotations(annotations)
+        elif meas_date == 'from_raw':
+            raw_er_prepared.set_meas_date(raw.info['meas_date'])
+            raw_er_prepared.set_annotations(raw.annotations)
+    elif annotations == 'keep':
+        if meas_date == 'keep':
+            pass  # no-op
+        elif meas_date == 'from_raw':
+            raw_er_prepared.set_meas_date(raw.info['meas_date'])
```

Comment:
I think some of this can be simplified. I went to do this and think I found some bug(s) with our annotations. I'll work on those first :(

Reply:
... over in https://github.com/mne-tools/mne-python/pull/10491

Reply:
Argh! Good luck! And yes, can certainly be largely improved, it was just a sketch of an idea :)